CMakeLists.txt

```diff
@@ -175,10 +175,22 @@ endif()
 option(MERCURY_ENABLE_DEBUG "Enable debug statements." OFF)
 if(MERCURY_ENABLE_DEBUG)
   set(HG_HAS_DEBUG 1)
+  set(HG_HAS_DIAG 1)
```

Comment:
I think here you should do instead (so that there is no conflict with the value of `HG_HAS_DIAG`):
```
  # Always enable counters if debug is turned ON
  set(MERCURY_ENABLE_COUNTERS "ON" CACHE BOOL "Enable diagnostics counters (enabled with debug)." FORCE)
```